### PR TITLE
Kamila/add search user fragment

### PR DESCRIPTION
**Pull request by kamilababayeva**
This PR add SearchUserActivity as a fragment

### CODE REVIEW

File: app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt [hunk 1]
```diff
@@ -24,6 +37,26 @@ class QrScanningActivityTest {
...
 
+    @BindValue
+    @JvmField
+    val mockDataGetter = mockk<DataGetter>()
+
+    private fun createMockDataGetter() {
+        val mockUser1 = User("mockUser1", uid = "mockUser1uid")
+        val mockUser2 = User("mockUser2", uid = "mockUser2uid")
+        val mockUser3 = User("mockUser3", uid = "mockUser3uid")
+        val mockUser = User("mockUser", uid = "mockUseruid", following = mapOf(Pair(mockUser2.uid,true), Pair(mockUser3.uid,true)))
```
Comment (MaximeZmt) [inline app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt:48]:
```suggestion
        val mockUser = User("mockUser", uid = "mockUseruid", following = mapOf(Pair(mockUser2.uid, true), Pair(mockUser3.uid, true)))

```

File: app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt [hunk 2]
```diff
@@ -24,6 +37,26 @@ class QrScanningActivityTest {
...
+
+        val mockUIDs = arrayListOf<String>("mockUser1uid","mockUser2uid","mockUser3uid")
```
Comment (MaximeZmt) [inline app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt:50]:
```suggestion
        val mockUIDs = arrayListOf<String>("mockUser1uid", "mockUser2uid", "mockUser3uid")

```

File: app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt [hunk 3]
```diff
@@ -45,23 +78,24 @@ class QrScanningActivityTest {
...
     @Test
     fun testNoExtra() {
         val ctx = ApplicationProvider.getApplicationContext() as Context
         val intent = Intent(ctx, QrScanningActivity::class.java)
+        createMockDataGetter()
 
-        val scn: ActivityScenario<ProfileActivity> = ActivityScenario.launch(intent)
-        Intents.intended(IntentMatchers.hasComponent(SearchUserActivity::class.java.name))
+        val scn: ActivityScenario<QrScanningActivity> = ActivityScenario.launch(intent)
+        assertThat(scn.result, hasResultCode(Activity.RESULT_CANCELED));
```
Comment (MaximeZmt) [inline app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt:96]:
```suggestion
        assertThat(scn.result, hasResultCode(Activity.RESULT_CANCELED))

```
why ";" ?

File: app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt [hunk 4]
```diff
@@ -134,15 +142,18 @@ class QrScanningActivity : AppCompatActivity() {
...
     }
 
-
-    private fun startActivityWExtra(intent: Intent, name: String?, arg: Serializable?) {
-        if (name != null && arg != null) {
-            intent.putExtra(name, arg)
-        }
+    private fun startActivityWExtra(intent: Intent, name: String, arg: Serializable) {
+        intent.putExtra(name, arg)
         startActivity(intent)
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt:147]: Now that we have an Intent Switching class, maybe you can refactor it.

Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt:147]: even delete this method and use directly the intent switching where it is called

**Review by MaximeZmt — COMMENTED** [review]
Nice work 👍 I have left some comments below

File: app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt (hunk 1 above)
Comment (kamilababayeva) [reply app/src/androidTest/java/ch/sdp/vibester/activity/QrScanningActivityTest.kt:48]: fixed

File: app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt (hunk 4 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt:147]: fixed!

**Review by MaximeZmt — APPROVED** [review]
Nice work, LGTM 👍

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 5]
```diff
@@ -47,34 +50,25 @@ class SearchUserActivity : AppCompatActivity(), OnItemClickListener {
...
         recyclerView!!.setHasFixedSize(true)
-        recyclerView!!.layoutManager = LinearLayoutManager(this)
+        recyclerView!!.layoutManager = LinearLayoutManager(requireContext())
 
         userProfileAdapter = UserProfileAdapter(this.users, authenticator, usersRepo, imageGetter, this)
 
         recyclerView!!.adapter = userProfileAdapter
 
-        searchEditText = findViewById(R.id.searchUserET)
+        searchEditText = view.findViewById(R.id.searchUserET)
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:64]: Is the "view" necessary here?

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 6]
```diff
@@ -47,34 +50,25 @@ class SearchUserActivity : AppCompatActivity(), OnItemClickListener {
...
         searchForUsers("")
 
-        val buttonScan: FloatingActionButton = findViewById(R.id.searchUser_scanning)
-
-        val extras = intent.extras
+        val buttonScan: FloatingActionButton = view.findViewById(R.id.searchUser_scanning)
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:67]: same for view

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 7]
```diff
@@ -87,6 +81,15 @@ class SearchUserActivity : AppCompatActivity(), OnItemClickListener {
         })
     }
 
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_search_user, container, false)
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:89]:
```suggestion
```

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 8]
```diff
@@ -87,6 +81,15 @@ class SearchUserActivity : AppCompatActivity(), OnItemClickListener {
...
+        return view
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:90]:
```suggestion
        return inflater.inflate(R.layout.fragment_search_user, container, false)
```

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt [hunk 9]
```diff
@@ -10,10 +10,7 @@ class IntentSwitcher private constructor() {
         fun switch(ctx: Context, arg: Class<*>?, extrasMap: Map<String, Serializable>? = null) {
             val intent = Intent(ctx, arg)
             intent.setFlags(FLAG_ACTIVITY_NEW_TASK)
-            if (extrasMap != null) {
-                extrasMap.forEach{
-                        entry -> intent.putExtra(entry.key, entry.value)
-                }
+            extrasMap?.forEach{ entry -> intent.putExtra(entry.key, entry.value)
             }
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:14]:
```suggestion
            

```

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt [hunk 10]
```diff
@@ -10,10 +10,7 @@ class IntentSwitcher private constructor() {
         fun switch(ctx: Context, arg: Class<*>?, extrasMap: Map<String, Serializable>? = null) {
             val intent = Intent(ctx, arg)
             intent.setFlags(FLAG_ACTIVITY_NEW_TASK)
-            if (extrasMap != null) {
-                extrasMap.forEach{
-                        entry -> intent.putExtra(entry.key, entry.value)
-                }
+            extrasMap?.forEach{ entry -> intent.putExtra(entry.key, entry.value)
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:13]:
```suggestion
            extrasMap?.forEach{ entry -> intent.putExtra(entry.key, entry.value) }

```

**Review by jiabaow — COMMENTED** [review]
A more general question, how many fragments will we have in total in the navigation bar and what are they? 
I feel like more than four will be too crowded, since we already have home, game, scoreboard.  But what about MyProfile, do you think it will be helpful to convert that to a fragment, too?

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt (hunk 5 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:64]: yes, cause it is fragment, and not an activity

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt (hunk 8 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:90]: fixed

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt (hunk 5 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:64]: fragments have different way to manage view, context and activity

**Comment by kamilababayeva** [comment]
> A more general question, how many fragments will we have in total in the navigation bar and what are they? I feel like more than four will be too crowded, since we already have home, game, scoreboard. But what about MyProfile, do you think it will be helpful to convert that to a fragment, too?

We were thinking to have 4 buttons in total: play, search, scoreboard and profile

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt [hunk 11]
```diff
@@ -47,34 +50,25 @@ class SearchUserActivity : AppCompatActivity(), OnItemClickListener {
...
         searchForUsers("")
 
-        val buttonScan: FloatingActionButton = findViewById(R.id.searchUser_scanning)
-
-        val extras = intent.extras
+        val buttonScan: FloatingActionButton = view.findViewById(R.id.searchUser_scanning)
 
         buttonScan.setOnClickListener {
-            val qrIntent = Intent(this, QrScanningActivity::class.java)
+            val qrIntent = Intent(requireActivity(), QrScanningActivity::class.java)
```
Comment (jiabaow) [inline app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:70]: Is it possible to test this?

**Review by jiabaow — APPROVED** [review]
LGTM

File: app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt (hunk 11 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt:70]: I think yes, wil fix it in the next PR!

**Pull request merged**